gems/smithy-client/spec/support/client_helper.rb

```diff
@@ -0,0 +1,121 @@
+        # },
+        'smithy.ruby.tests#Enum' => {
+          'type' => 'enum',
+          'members' => {
+            'member' => {
+              'target' => 'smithy.api#Unit',
+              'traits' => { 'smithy.api#enumValue' => 'value' }
+            }
+          }
+        },
+        'smithy.ruby.tests#intEnum' => {
+          'type' => 'intEnum',
+          'members' => {
+            'member' => {
+              'target' => 'smithy.api#Unit',
+              'traits' => { 'smithy.api#enumValue' => 1 }
+            }
+          }
+        },
+        'smithy.ruby.tests#List' => {
+          'type' => 'list',
+          'member' => { 'target' => 'smithy.api#String' }
+        },
+        'smithy.ruby.tests#Map' => {
+          'type' => 'map',
+          'key' => { 'target' => 'smithy.api#String' },
+          'value' => { 'target' => 'smithy.api#String' }
+        },
+        'smithy.ruby.tests#Structure' => {
+          'type' => 'structure',
+          'members' => {
+            'bigDecimal' => { 'target' => 'smithy.api#BigDecimal' },
+            'bigInteger' => { 'target' => 'smithy.api#BigInteger' },
+            'blob' => { 'target' => 'smithy.api#Blob' },
+            'boolean' => { 'target' => 'smithy.api#Boolean' },
+            'byte' => { 'target' => 'smithy.api#Byte' },
+            # 'document' => { 'target' => 'smithy.api#Document' },
+            'double' => { 'target' => 'smithy.api#Double' },
+            'enum' => { 'target' => 'smithy.ruby.tests#Enum' },
+            'float' => { 'target' => 'smithy.api#Float' },
+            'integer' => { 'target' => 'smithy.api#Integer' },
+            'intEnum' => { 'target' => 'smithy.ruby.tests#intEnum' },
+            'long' => { 'target' => 'smithy.api#Long' },
+            'short' => { 'target' => 'smithy.api#Short' },
+            'string' => { 'target' => 'smithy.api#String' },
+            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
+            'structure' => { 'target' => 'smithy.ruby.tests#Structure' },
+            'list' => { 'target' => 'smithy.ruby.tests#List' },
+            'map' => { 'target' => 'smithy.ruby.tests#Map' }
+            # 'union' => { 'target' => 'smithy.api#String' }
+          }
+        },
+        'smithy.ruby.tests#Operation' => {
+          'type' => 'operation',
+          'input' => { 'target' => 'smithy.ruby.tests#Structure' },
+          'output' => { 'target' => 'smithy.ruby.tests#Structure' }
+        },
+        'smithy.ruby.tests#SampleClient' => {
+          'type' => 'service',
+          'operations' => [
+            { 'target' => 'smithy.ruby.tests#Operation' }
+          ]
+        }
+      }
+    end
+
+    def sample_client(options = {})
+      model = options[:model] ||= model(options)
+      namespace = options[:gem_module] || next_sample_module_name
+      plan = create_plan(model, namespace, options)
+      source = Smithy.source(plan)
+      Object.module_eval(source)
```

Comment:
Since we are doing the eval on `Object` we are adding this to the global namespace.  We should add functions for cleaning up and managing these as well I think.

Reply:
Sure

Reply:
As opposed to files, these will get garbage collected (next sample name will ensure they are unique). I will further refactor this and spec_helper to use something common but I will do it outside of this PR

Reply:
I dont think garbage collection works on modules/classes that get defined?

Reply:
I will check, but I will also add a clean up when we have a test that uses the source. I will do some follow up changes.